### PR TITLE
Upgrade admin-bundle to ~1.1@dev

### PR DESCRIPTION
**Pull request by rn0**

### CODE REVIEW

File: Doctrine/Admin/Context/ResourceRepositoryContext.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public function __construct($requestHandlers, ResourceFormBuilder $resourceFormBuilder, LocaleManager $localeManager)
+    {
+        parent::__construct($requestHandlers, $resourceFormBuilder);
+        $this->localeManager = $localeManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData()
```
Comment (bartekgruszka) [inline Doctrine/Admin/Context/ResourceRepositoryContext.php]: This function is duplicated in `BatchElementContext`

File: Doctrine/Admin/TranslatableResourceElement.php [hunk 2]
```diff
@@ -21,7 +28,7 @@ public function setLocaleManager(LocaleManager $localeManager)
     }
 
     /**
-     * {@inheritdoc}
+     * {@inheritdoc}l
```
Comment (bartekgruszka) [inline Doctrine/Admin/TranslatableResourceElement.php]: ?

File: Resources/views/List/list.html.twig [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        {% if element.option('allow_add') == true %}
+            <a href="{{ path('fsi_admin_translatable_form', {element : element.id, locale: translatable_locale, redirect_uri: app.request.uri}) }}" class="btn btn-success" id="create-element">
+                <span class="glyphicon glyphicon-plus"></span>
+                {{ 'crud.list.button.create'|trans({}, 'FSiAdminBundle') }}
+            </a>
+        {% endif %}
+    {% endblock buttons %}
+{% endblock bottom_buttons %}
+
+{#{% block javascripts %}#}
```
Comment (bartekgruszka) [inline Resources/views/List/list.html.twig]: Should be removed?

File: features/fixtures/project/app/AppKernel.php [hunk 4]
```diff
@@ -29,13 +29,13 @@ public function registerContainerConfiguration(LoaderInterface $loader)
...
-    public function getCacheDir()
-    {
-        return sys_get_temp_dir() . '/FSiAdminTranslatableBundle/cache';
-    }
-
-    public function getLogDir()
-    {
-        return sys_get_temp_dir() . '/FSiAdminTranslatableBundle/logs';
-    }
+//    public function getCacheDir()
```
Comment (bartekgruszka) [inline features/fixtures/project/app/AppKernel.php]: ?

File: features/fixtures/project/web/app_test.php [hunk 5]
```diff
@@ -3,15 +3,22 @@
...
+
 Debug::enable();
 
 require_once __DIR__.'/../app/AppKernel.php';
 
-$kernel = new AppKernel('test', true);
-$kernel->loadClassCache();
-Request::enableHttpMethodParameterOverride();
+// Initialize kernel and run the application.
+$kernel = new AppKernel('dev', true);
```
Comment (bartekgruszka) [inline features/fixtures/project/web/app_test.php]: File name is `app_test.php` but symfony env is set to `dev`

File: Controller/TranslatableBatchController.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+namespace FSi\Bundle\AdminTranslatableBundle\Controller;
+
+use FSi\Bundle\AdminBundle\Admin\CRUD\BatchElement;
+use FSi\Bundle\AdminBundle\Controller\BatchController as BaseBatchController;
+use FSi\Bundle\AdminBundle\Exception\ContextException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class TranslatableBatchController extends BaseBatchController
```
Comment (chives) [inline Controller/TranslatableBatchController.php]: If anyone can imagine any use case for "translatable" batch action please let us know :)

File: Controller/TranslatableFormController.php [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+use FSi\Bundle\AdminBundle\Admin\CRUD\FormElement;
+use FSi\Bundle\AdminBundle\Controller\FormController as BaseFormController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class TranslatableFormController extends BaseFormController
+{
+    public function formAction(FormElement $element, Request $request)
+    {
+        $context = $this->contextManager->createContext('fsi_admin_translatable_form', $element);
```
Comment (chives) [inline Controller/TranslatableFormController.php]: we can use ``handleRequest()`` method now https://github.com/fsi-open/admin-bundle/blob/master/Controller/ControllerAbstract.php#L58

File: Controller/TranslatableBatchController.php (hunk 6 above)
Comment (bartekgruszka) [inline Controller/TranslatableBatchController.php]: Maybe something like 'remove translation'?

Comment (chives) [inline Controller/TranslatableBatchController.php]: ok, let's leave it as is

**Comment by rn0** [comment]
@fsi-open

File: Resources/views/CRUD/datagrid.html.twig [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{% extends '@FSiAdmin/List/datagrid.html.twig' %}
```
Comment (chives) [inline Resources/views/CRUD/datagrid.html.twig]: this template is useless. we should also test inplace edit at datagrid in this bundle

**Comment by rn0** [comment]
ToDo:
* [ ] inline edit on list

File: EventListener/ToolsMenuListener.php [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+    private function populateTranslationLocaleMenu(Item $menu)
+    {
+        $requestParameters = $this->getRequestParameters();
+        $route = $this->request->get('_route');
+
+        foreach ($this->localeManager->getLocales() as $locale) {
+            $requestParameters['locale'] = $locale;
+
+            $localeItem = new RoutableItem(sprintf('translation-locale.%s', $locale), $route, $requestParameters);
```
Comment (chives) [inline EventListener/ToolsMenuListener.php]: we could use ``Symfony\Component\Intl\Intl::getLanguageBundle()->getLanguageName()`` to pass localized language name
https://github.com/symfony/Intl/blob/818db47dd789f60f6af8773d880ea184f7e291fc/ResourceBundle/LanguageBundle.php#L57

**Comment by rn0** [comment]
@fsi-open

**Comment by rn0** [comment]
@fsi-open

**Comment by rn0** [comment]
@fsi-open

**Comment by chives** [comment]
:+1:

**Comment by bartekgruszka** [comment]
:+1:

**Comment by JarekW** [comment]
:+1:

**Pull request merged**